Documentation/api.md

```diff
@@ -1,5 +1,27 @@
 # etcd API
 
+- [Key Space Operations](#key-space-operations)
```

Comment:
I've combined the directory operations, but I think we need to spit this doc up. Consider this the first step in improving readability.